src/scitacean/_base_model.py

```diff
@@ -0,0 +1,211 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 SciCat Project (https://github.com/SciCatProject/scitacean)
+
+"""Types and functions to implement models for communication with SciCat."""
+
+try:
+    # Python 3.11+
+    from enum import StrEnum as _StrEnum
+
+    _DatasetTypeBases = (_StrEnum,)
+except ImportError:
+    from enum import Enum as _Enum
+
+    _DatasetTypeBases = (
+        str,
+        _Enum,
+    )
+
+import dataclasses
+from typing import Any, Dict, Iterable, Optional, Type, TypeVar
+
+import pydantic
+
+from ._internal.orcid import is_valid_orcid
+from .filesystem import RemotePath
+from .logging import get_logger
+from .pid import PID
+
+ModelType = TypeVar("ModelType", bound=pydantic.BaseModel)
+
+
+class DatasetType(*_DatasetTypeBases):
+    """Type of Dataset."""
+
+    RAW = "raw"
+    DERIVED = "derived"
+
+
+class BaseModel(pydantic.BaseModel):
+    """Base class for Pydantic models for communication with SciCat."""
+
+    class Config:
+        extra = pydantic.Extra.forbid
+        json_encoders = {
+            PID: lambda v: str(v),
+            RemotePath: lambda v: v.posix,
+        }
+
+    # Some schemas contain fields that we don't want to use in Scitacean.
+    # Normally, omitting them from the model would result in an error when
+    # building a model from the JSON returned by SciCat.
+    # The following subclass hook allows models to mark fields as masked.
+    # Those will be silently dropped by __init__.
+    # Note also the comment for _IGNORED_KWARGS below.
+    def __init_subclass__(
+        cls, /, masked: Optional[Iterable[str]] = None, **kwargs: Any
+    ) -> None:
+        super().__init_subclass__(**kwargs)
+
+        masked = list(masked) if masked is not None else []
+        field_names = {field.alias for field in cls.__fields__.values()}
+        masked.extend(key for key in _IGNORED_KWARGS if key not in field_names)
```

Comment:
It can happen. This all depends on `_IGNORED_KWARGS` and `tools/model-generation/spec/masked-fields.yml`. But it wouldn't matter with the way `_delete_ignored_args` is implemented.